csi.proto

```diff
@@ -1206,6 +1206,10 @@ message NodeStageVolumeRequest {
   // CO SHALL be responsible for creating the directory if it does not
   // exist.
   // This is a REQUIRED field.
+  // This field overrides the general CSI size limit.
+  // The size of this field SHALL NOT exceed 256 bytes. The general
+  // CSI size limit, 128 byte, is RECOMMENDED for best backwards
+  // compatibility.
```

Comment:
I was checking the systemd issue again and 256 indeed is too big to be a unit name and hence mount point that is managed by systemd. More details are here - https://github.com/systemd/systemd/pull/15911/files#diff-cba2cdf3ac7ac3975d69ed999252a138fed94763e46312c7e48500744eb8028aR452 . The issue itself is discussed in more  detail - https://github.com/systemd/systemd/pull/14294

The underlying reasoning is - on Linux `FILENAME_MAX` is 256 chars and hence you can't make a file whose name is bigger than 256 character. Now each mount point typically when managed via systemd gets converted to a unit name. This conversion happens on *whole* path and hence if it is bigger than 256 chars, systemd fails to manage it.

It should also be noted that, there is another variable called `PATH_MAX` which defines the maximum length of whole path (i.e not just the last component) and according to https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation the value is 260 characters on windows and hence setting this to 256 char will be pushing it awfully close to the upper limit.

So I think that 256 char might actually be too close to the upper limit of maximum path lengths and we might have to reduce the length. I don't know what good default will be but something smaller I think.

Reply:
Hmm, what about 192? Seems like a magic number to use

Reply:
I am putting 192 as the reduced length for both node_id and path. So it could be less error-prone.

We can talk more in the next CSI meeting to confirm this tho.